extension/source/background/AggregatorController.ts

```diff
@@ -158,4 +175,55 @@ export default class AggregatorController {
       this.preferredAggregators[providerId] = preferredAggregator;
     },
   });
+
+  async addFeePaymentAction(
+    nonce: string,
+    actions: Array<ActionData>,
+    wallet: BlsWalletWrapper,
+    aggregator: Aggregator,
+  ) {
+    const network = await this.networkController.network.read();
+    const netCfg = getNetworkConfig(network, this.multiNetworkConfig);
+    const ethersProvider = await this.ethersProvider.read();
+
+    const aggregatorUtilitiesAddress = netCfg.addresses.utilities;
+    const aggregatorUtilitiesContract = AggregatorUtilitiesFactory.connect(
+      aggregatorUtilitiesAddress,
+      ethersProvider,
+    );
+
+    const actionsWithFeePaymentAction = [
+      ...actions,
+      {
+        ethValue: 1,
+        contractAddress: aggregatorUtilitiesAddress,
+        encodedFunction:
+          aggregatorUtilitiesContract.interface.encodeFunctionData(
+            'sendEthToTxOrigin',
+          ),
+      },
+    ];
+
+    const feeEstimateBundle = wallet.sign({
+      nonce,
+      actions: [...actionsWithFeePaymentAction],
+    });
+
+    const feeEstimate = await aggregator.estimateFee(feeEstimateBundle);
+    const feeRequired = BigNumber.from(feeEstimate.feeRequired);
+    const safetyPremium = feeRequired.div(5);
```

Comment:
Do you think there would be a reason to change this `safetyPremium` semi frequently?  Just wondering if we should make it an env variable

Reply:
Yeah I think that's a good question. I originally took the value of 5 from [an aggregator test](https://github.com/web3well/bls-wallet/blob/70ca00f0894b301c24692e3c5aa13efebd69c9e1/aggregator/test/AggregationStrategy.test.ts#L76), and used that in the provider. I'm not sure on the answer to the question tbh, and would be interested to get Andrews thoughts on this. I think another useful question would be is 20% to high as a safety margin? I've noticed the new `signWithGasEstimate` method in the contract-updates has a default value of 0, so after seeing that, a margin of 20% here seems very high.

@voltrevo do you see the need for a value like this to be changed ever?  

And is a 20% margin even sensible in this case?

Reply:
Decided to merge btw as wanted to add this change to `merge-in-contract-updates`, think this change would be useful to test quill with, as we want to enforce paying fees as the default behaviour. 

Can add follow ups if needed

Reply:
Yeah 20% is quite high. I think it's fine for now so that we err on the side of reliability, but we'll definitely want to scrutinize this once compression delivers competitive pricing.

Some thoughts off the top of my head:
- Add this to user configuration
- Aggregator could estimate how much each account has over-paid and adjust accordingly

Reply:
Will add the safetyPremium to extension config.

> Aggregator could estimate how much each account has over-paid and adjust accordingly

This sounds useful, worth creating an issue? Or just to be mindful of this when we circle back following compression work?